src/app/components/Statistics/ChartConfig/ChartConfig.ts

```diff
@@ -0,0 +1,252 @@
+            case typesStatistic.UNIQUE_ADRESSES:
+                this.data = dataChart.addressInfo
+                break
+            case typesStatistic.DAILY_TOKEN_TRANSFER:
+                this.data = dataChart
+                break
+            case typesStatistic.GAS_USED:
+                this.data = dataChart.txInfo
+                break
+            case typesStatistic.ACTIVE_ADDRESSES:
+                this.data = dataChart.addressInfo
+                break
+            case typesStatistic.GAS_AVERAGE_PRICE:
+                this.highestAndLowestInfo = {
+                    highestValue: dataChart.highestValue,
+                    highestDate: dataChart.highestDate,
+                    lowerValue: dataChart.lowerValue,
+                    lowerDate: dataChart.lowerDate,
+                }
+                this.data = dataChart.txInfo
+                break
+            case typesStatistic.GAS_AVERAGE_LIMIT:
+                this.data = dataChart
+                break
+            case typesStatistic.AVERAGE_BLOCK_SIZE:
+                this.data = dataChart
+                break
+            case typesStatistic.CO2_EMISSIONS:
+                this.data = dataChart
+                break
+        }
+
+        this.checkRangeTimeString(timeSeeAxis)
+    }
+
+    public getCategories() {
+        switch (this.typeChartData) {
+            case typesStatistic.DAILY_TRANSACTIONS:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.date))
+            case typesStatistic.UNIQUE_ADRESSES:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.dateAt))
+            case typesStatistic.GAS_USED:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.date))
+            case typesStatistic.ACTIVE_ADDRESSES:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.dateAt))
+            case typesStatistic.AVERAGE_BLOCK_SIZE:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.dateInfo))
+            case typesStatistic.GAS_AVERAGE_PRICE:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.date))
+            case typesStatistic.DAILY_TOKEN_TRANSFER:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.dateAt))
+            case typesStatistic.CO2_EMISSIONS:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.time))
+            default:
+                return this.data.map((value, index) => this.getCheckTimeCategories(value.Date))
+        }
+    }
+
+    private checkRangeTimeString(timeSeeAxis: String) {
+        console.log(timeSeeAxis)
+        if (timeSeeAxis === 'custom') {
+            this.timeSeeAxis = this.verifyRangeTime()
+        } else {
+            this.timeSeeAxis = timeSeeAxis
+        }
+    }
+
+    private verifyRangeTime() {
```

Comment:
im not sure about the initial date can we have something close to the current date also is there any option to switch the year without spamming the left/right arrow